2000-09-10  Anthony Green  <green@redhat.com>

	* class.c (layout_class): Handle case where superclass can't be
	layed out yet.

*** gcc/java/class.c	2000/08/11 22:01:37	1.74
--- gcc/java/class.c	2000/09/11 00:43:33
*************** layout_class (this_class)
*** 1811,1818 ****
  
    if (super_class)
      {
!       super_class = maybe_layout_super_class (super_class, this_class);
!       if (TREE_CODE (TYPE_SIZE (super_class)) == ERROR_MARK)
  	{
  	  TYPE_SIZE (this_class) = error_mark_node;
  	  CLASS_BEING_LAIDOUT (this_class) = 0;
--- 1811,1820 ----
  
    if (super_class)
      {
!       tree maybe_super_class 
! 	= maybe_layout_super_class (super_class, this_class);
!       if (maybe_super_class == NULL
! 	  || TREE_CODE (TYPE_SIZE (maybe_super_class)) == ERROR_MARK)
  	{
  	  TYPE_SIZE (this_class) = error_mark_node;
  	  CLASS_BEING_LAIDOUT (this_class) = 0;
